### ISSUE TITLE
PReLULayer cannot be cast to org.deeplearning4j.nn.conf.layers.FeedForwardLayer

### ISSUE THREAD

**Issue by xiptos** [deeplearning4j/deeplearning4j#6257]
I believe this is associated with the issue #6245.
Using the latest snapshot, I'm getting the following exception:

08:26:16.418 [main] INFO org.deeplearning4j.nn.modelimport.keras.Hdf5Archive - Unexpected end-of-input: was expecting closing '"' for name
 at [Source: {"class_name": "Model", "config": {"layers": [{"class_name": "InputLayer", "config": {"dtype": "float32", "batch_input_shape": [null, null, null, 3], "name": "input_3", "sparse": false}, "inbound_nodes": [], "name": "input_3"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv1", "bias_regularizer": null, "bias_constraint": null, "activation": "linear", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 10, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3, 3]}, "inbound_nodes": [[["input_3", 0, 0, {}]]], "name": "conv1"}, {"class_name": "PReLU", "config": {"name": "PReLU1", "alpha_constraint": null, "trainable": true, "alpha_regularizer": null, "alpha_initializer": {"class_name": "Zeros", "config": {}}, "shared_axes": [1, 2]}, "inbound_nodes": [[["conv1", 0, 0, {}]]], "name": "PReLU1"}, {"class_name": "MaxPooling2D", "config": {"name": "max_pooling2d_6", "trainable": true, "data_format": "channels_last", "pool_size": [2, 2], "padding": "valid", "strides": [2, 2]}, "inbound_nodes": [[["PReLU1", 0, 0, {}]]], "name": "max_pooling2d_6"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv2", "bias_regularizer": null, "bias_constraint": null, "activation": "linear", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 16, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3, 3]}, "inbound_nodes; line: 1, column: 4001]
08:26:16.424 [main] INFO org.deeplearning4j.nn.modelimport.keras.Hdf5Archive - Unexpected end-of-input: was expecting closing quote for a string value
 at [Source: {"class_name": "Model", "config": {"layers": [{"class_name": "InputLayer", "config": {"dtype": "float32", "batch_input_shape": [null, null, null, 3], "name": "input_3", "sparse": false}, "inbound_nodes": [], "name": "input_3"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv1", "bias_regularizer": null, "bias_constraint": null, "activation": "linear", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 10, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3, 3]}, "inbound_nodes": [[["input_3", 0, 0, {}]]], "name": "conv1"}, {"class_name": "PReLU", "config": {"name": "PReLU1", "alpha_constraint": null, "trainable": true, "alpha_regularizer": null, "alpha_initializer": {"class_name": "Zeros", "config": {}}, "shared_axes": [1, 2]}, "inbound_nodes": [[["conv1", 0, 0, {}]]], "name": "PReLU1"}, {"class_name": "MaxPooling2D", "config": {"name": "max_pooling2d_6", "trainable": true, "data_format": "channels_last", "pool_size": [2, 2], "padding": "valid", "strides": [2, 2]}, "inbound_nodes": [[["PReLU1", 0, 0, {}]]], "name": "max_pooling2d_6"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv2", "bias_regularizer": null, "bias_constraint": null, "activation": "linear", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 16, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3, 3]}, "inbound_nodes": [[["max_pooling2d_6", 0, 0, {}]]], "name": "conv2"}, {"class_name": "PReLU", "config": {"name": "PReLU2", "alpha_constraint": null, "trainable": true, "alpha_regularizer": null, "alpha_initializer": {"class_name": "Zeros", "config": {}}, "shared_axes": [1, 2]}, "inbound_nodes": [[["conv2", 0, 0, {}]]], "name": "PReLU2"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv3", "bias_regularizer": null, "bias_constraint": null, "activation": "linear", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 32, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [3, 3]}, "inbound_nodes": [[["PReLU2", 0, 0, {}]]], "name": "conv3"}, {"class_name": "PReLU", "config": {"name": "PReLU3", "alpha_constraint": null, "trainable": true, "alpha_regularizer": null, "alpha_initializer": {"class_name": "Zeros", "config": {}}, "shared_axes": [1, 2]}, "inbound_nodes": [[["conv3", 0, 0, {}]]], "name": "PReLU3"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "VarianceScaling", "config": {"distribution": "uniform", "scale": 1.0, "seed": null, "mode": "fan_avg"}}, "name": "conv4-1", "bias_regularizer": null, "bias_constraint": null, "activation": "softmax", "trainable": true, "data_format": "channels_last", "padding": "valid", "strides": [1, 1], "dilation_rate": [1, 1], "kernel_regularizer": null, "filters": 2, "bias_initializer": {"class_name": "Zeros", "config": {}}, "use_bias": true, "activity_regularizer": null, "kernel_size": [1, 1]}, "inbound_nodes": [[["PReLU3", 0, 0, {}]]], "name": "conv4-1"}, {"class_name": "Conv2D", "config": {"kernel_constraint": null, "kernel_initializer": {"class_name": "Var; line: 1, column: 8001]
08:26:16.584 [main] INFO org.nd4j.linalg.factory.Nd4jBackend - Loaded [CpuBackend] backend
08:26:17.228 [main] INFO org.nd4j.nativeblas.NativeOpsHolder - Number of threads used for NativeOps: 2
08:26:17.486 [main] INFO org.nd4j.nativeblas.Nd4jBlas - Number of threads used for BLAS: 2
08:26:17.609 [main] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Backend used: [CPU]; OS: [Mac OS X]
08:26:17.609 [main] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Cores: [4]; Memory: [1.8GB];
08:26:17.609 [main] INFO org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner - Blas vendor: [OPENBLAS]
08:26:17.765 [main] INFO org.deeplearning4j.nn.graph.ComputationGraph - Starting ComputationGraph with WorkspaceModes set to [training: ENABLED; inference: ENABLED], cacheMode set to [NONE]
Exception in thread "main" java.lang.ClassCastException: org.deeplearning4j.nn.conf.layers.PReLULayer cannot be cast to org.deeplearning4j.nn.conf.layers.FeedForwardLayer
	at org.deeplearning4j.nn.graph.ComputationGraph.summary(ComputationGraph.java:4187)
	at pt.ipb.face.Mtcnn2Dl4j.main(Mtcnn2Dl4j.java:26)

**Comment by maxpumperla** [comment]
@xiptos right, good point. note that this "only" affects your call to `summary()`, nothing else. DL4J networks try to cast to FeedForward layer when summarizing the network. Will take care of this.
